sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/view/AttributesProcessor.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.view;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.BiFunction;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * An {@code AttributesProcessor} is used by {@code View}s to define the actual recorded set of
+ * attributes.
+ *
+ * <p>An AttributesProcessor is used to define the actual set of attributes that will be used in a
+ * Metric vs. the inbound set of attributes from a measurement.
+ */
+@Immutable
+public abstract class AttributesProcessor {
+  private AttributesProcessor() {}
+
+  /**
+   * Manipulates a set of attributes, returning the desired set.
+   *
+   * @param incoming Attributes associated with an incoming measurement.
+   * @param context The context associated with the measurement.
+   */
+  public abstract Attributes process(Attributes incoming, Context context);
+
+  /**
+   * If true, this ensures the `Context` argument of the attributes processor is always accurate.
+   * This will prevents bound instruments from pre-locking their metric-attributes and defer until
+   * context is available.
+   */
+  public abstract boolean usesContext();
```

Comment:
Is this safe enough? Couldn't implementations be adding conditionally adding attributes from other things (like a pre-configured Resource, for example)?

Reply:
In fact, isn't it the case that *any* dynamic manipulation of attributes will break the binding? Maybe we need this to be "isDynamic()" or something?

Reply:
Yeah, this could be better documented.  One reason I tried not to expose this interface any longer was so only the extensions we provide are allowed.

To the extent this was intended to be a "pure functional" API, the only input that matters is context.  Happy to figure out how to document and change this portion.